Verify end of tour date not in past

evdhoudt opened the pull request:
Closes NCI-Agency/anet#1367

gjvoosten commented on client/src/models/Person.js:
```diff
@@ -128,6 +129,17 @@ export default class Person extends Model {
+                "end-of-tour-date",
+                `The ${
+                  Settings.fields.person.endOfTourDate
+                } date must be set in the future`,
```
Change *must be set in the future* to *must be in the future*.

gjvoosten commented on client/src/models/Person.js:
```diff
@@ -34,6 +34,7 @@ export default class Person extends Model {
   static yupSchema = yup
     .object()
     .shape({
+      uuid: yup.string().default(""),
```
Use `yup.string().nullable().default(null)` (a new object should have a null uuid, not the empty string).

gjvoosten commented on client/src/models/Person.js:
```diff
@@ -128,6 +129,17 @@ export default class Person extends Model {
               )
             : schema.nullable()
         )
+        .when(["role", "status", "uuid"], (role, status, uuid, schema) =>
```
Why not combine this with the `.when` directly above it?

evdhoudt commented:
> Also run `yarn run lint-fix` before pushing to make sure code is properly formatted.

Hmm... does this mean that I cannot use vscode for formatting? Both prettier and eslint reformats the file in other ways than our linter.

gjvoosten commented on client/src/models/Person.js:
```diff
@@ -34,6 +34,7 @@ export default class Person extends Model {
   static yupSchema = yup
     .object()
     .shape({
+      uuid: yup.string().nullable().default(null),
```
`yarn run lint-fix`

gjvoosten commented on client/src/models/Person.js:
```diff
@@ -119,15 +120,24 @@ export default class Person extends Model {
       endOfTourDate: yupDate
         .nullable()
-        .when("role", (role, schema) =>
+        .when(["role", "status", "uuid"], (role, status, uuid, schema) => {
```
You're not returning anything (should return the schema). Conditions can be simplified, I'll have a go at it.

evdhoudt commented:
Do we need a second opinion or can this PR be merged?

VassilIordanov commented:
> Do we need a second opinion or can this PR be merged?

My bad, apologies for the delay.. PR is good but the requirement was not fully flushed. I've added this to the requirement:
https://github.com/NCI-Agency/anet/issues/1367#issuecomment-489062725

evdhoudt commented:
@VassilIordanov @gjvoosten I've fixed the EOT warning.

It only shows when creating a new user during the on-boarding proces.
Otherwise a warning is shown when editing or creating a person with EOT in the past.

Also added wdio tests to check the validation during on-boarding.
